File: 현호/BOJ_1074.js
const fs = require("fs");
const input = fs.readFileSync("/dev/stdin").toString().trim().split("\n");

const [n, r, c] = input[0].split(" ").map(Number);

const solve = (n, r, c) => {
  if (n === 0) {
    return 0;
  }

  const half = 1 << (n - 1);

  if (r < half && c < half) {
    return solve(n - 1, r, c);
  } else if (r < half && c >= half) {
    return half * half + solve(n - 1, r, c - half);
  } else if (r >= half && c < half) {
    return 2 * half * half + solve(n - 1, r - half, c);
  } else {
    return 3 * half * half + solve(n - 1, r - half, c - half);
  }
};

console.log(solve(n, r, c));

File: 근재/Recursion/BOJ_1074.py
N, r, c = list(map(int, input().split()))
def zfunction(N, r, c):
    if N == 0:
        return 0
    return 2*(r%2)+(c%2) + 4*zfunction(N-1,int(r/2),int(c/2))

print(zfunction(N, r, c))
